test/ctia/entity/asset_test.clj

```diff
@@ -0,0 +1,92 @@
+(ns ctia.entity.asset-test
+  (:require [clj-momo.test-helpers.core :as mth]
+            [clojure.test :refer [deftest is are join-fixtures testing use-fixtures]]
+            [ctia.entity.asset :as asset]
+            [ctia.test-helpers.aggregate :as aggregate]
+            [ctia.test-helpers.auth :as auth]
+            [ctia.test-helpers.core :as helpers]
+            [ctia.test-helpers.crud :refer [entity-crud-test]]
+            [ctia.test-helpers.fake-whoami-service :as whoami-helpers]
+            [ctia.test-helpers.field-selection :as field-selection]
+            [ctia.test-helpers.http :as http]
+            [ctia.test-helpers.pagination :as pagination]
+            [ctia.test-helpers.store :as store]
+            [ctim.examples.assets :refer [new-asset-minimal new-asset-maximal]]))
+
+(use-fixtures :once (join-fixtures [mth/fixture-schema-validation
+                                    helpers/fixture-properties:clean
+                                    whoami-helpers/fixture-server]))
+
+(use-fixtures :each whoami-helpers/fixture-reset-state)
+
+(defn additional-tests [asset-id asset-sample]
+  (testing "GET /ctia/asset/search"
+    ;; only when ES store
+    (when (= "es" (get-in @ctia.properties/properties [:ctia :store :asset]))
+      (are [term check-fn expected desc] (let [response (helpers/get (str "ctia/asset/search")
+                                                                     :query-params {"query" term}
+                                                                     :headers {"Authorization" "45c1f5e3f05d0"})]
+                                           (is (= 200 (:status response)))
+                                           (is (= expected (check-fn response)) desc))
+
+        "asset_type:\"device\""
+        (fn [r] (-> r :parsed-body first :asset_type))
+        (-> asset-sample :asset_type)
+        "Searching by an Asset type works"
+
+        ;; TODO: Add more cases
+
+        ))))
+
+(deftest asset-routes-test
+  (store/test-for-each-store
+   (fn []
+     (helpers/set-capabilities! "foouser" ["foogroup"] "user" auth/all-capabilities)
+     (whoami-helpers/set-whoami-response http/api-key "foouser" "foogroup" "user")
+     (entity-crud-test
+      {:entity           "asset"
+       :example          new-asset-maximal
+       :invalid-tests?   false
+       :update-tests?    true
+       :search-tests?    false
```

Comment:
is there a specific reason why you disabled some tests?

Reply:
No... I think that's particular piece is the result of copy-pasta of some other test :) I'm still learning how things work. I'll try to enable them and see how it goes. Thank you for pointing that out! 👍